### ISSUE TITLE
Doesn't work with Visual Studio 2017 Update 3 - Open Folder / CMake support

### ISSUE THREAD

**Issue by rkeithhill** [csoltenborn/GoogleTestAdapter#156]
GTA works fine when we use the "Visual Studio 14 2015 Win64" generator and then open the generated SLN in VS 2015.  However, when we try to use the VS 2017 Update 3 support for open a folder that contains a CMake file (and configuring this feature to use the `Ninja` generator via the CMakeSettings.json file), GTA doesn't find any tests to execute.  Is this scenario supposed to work?

**Comment by csoltenborn** [comment]
No, unfortunately not. Note however that we are collaborating with MS for a couple of month now, and Open Folder might be added at some point in time (if it's technically feasible - I don't know whether the VS test adapter framework supports this), but there are no concrete plans at the moment. Let's still keep this as a reminder (so to say :-) ).

**Comment by rkeithhill** [comment]
Thanks for confirming.  If this support required a settings file to "direct" GTA to the exe/folder/CMakeLists.txt with the tests, I'd be happy to supply such a file e.g. `GoogleTestSettings.json`.  We already have to do that for the CMake support via a `CMakeSettings.json` file.

**Comment by LukaszMendakiewicz** [comment]
It is currently not possible to provide such support in the test adapter (GTA/TAfGT) due to the test platform (Test Explorer) not supporting Open Folder. This is something we (Microsoft) are looking to have added, but there is no ETA currently we can share.

**Comment by csoltenborn** [comment]
@LukaszMendakiewicz Thanks for clarifying this!

@rkeithhill In case this helps: GTA/TAfGT should work fine with makefiles (see e.g. #86).

**Comment by dawikur** [comment]
Any update on this?
On simple project, in the hierarchy tree I see one test, 
which was registered to cmake by `add_test`.
It passes/fails correctly, but it doesn't discover any suites.tests.
After adding '--verbose' to `CMakeSettings.json` in `CTest Command Arguments`
the output correctly shows standard gtest RUN-OK so it seems feasible.

**Comment by jacobq** [comment]
Is there somewhere (else) where the status of this issue is being tracked? It has been nearly 6 years since this was last updated. [CLion](https://www.jetbrains.com/clion/) supports [this feature out of the box now](https://www.jetbrains.com/help/clion/creating-google-test-run-debug-configuration-for-test.html), but I'd love to be able to do this with Visual Studio as well. Are you seeking donations? Maintainers? Hope you are well & thank you for your work thus far.

**Comment by csoltenborn** [comment]
Thanks for asking! I must admit that I stopped working on this project since I'm not working with C++ any more (thanks god ;-) ), and there's the MS clone (Test Adapter for Google Test) which is still developed by MS (and also supports VS2022). Thus, yes: I'm basically looking for a maintainer. I will update the readme accordingly in the coming days. Are you interested by any chance? I would certainly try to help you get started!

**Comment by jacobq** [comment]
> Thanks for asking! I must admit that I stopped working on this project since I'm not working with C++ any more (thanks god ;-) ), and there's the MS clone (Test Adapter for Google Test) which is still developed by MS (and also supports VS2022). Thus, yes: I'm basically looking for a maintainer. I will update the readme accordingly in the coming days. Are you interested by any chance? I would certainly try to help you get started!

Thank you for replying. Unfortunately, I am not an expert with C++ nor do I have the bandwidth to undertake becoming a maintainer of this work (can only handle doing employer-sponsored projects in my current life situation). However, it is helpful to know the project status though as I am exploring and evaluating tools for my team of embedded C developers to use for testing and automation. Thank you for updating the Readme / docs.
